src/key_value_store.rs

```diff
@@ -237,15 +242,15 @@ impl<K: Copy + Eq + Hash + Ord, V: Clone + Eq + Hash> Persist<(K, V)>
     }
 
     fn remove(&mut self, change: (K, V)) {
-        self.index.remove(&change.0);
+        self.index.entry(change.0).or_default().remove(&change.1);
```

Comment:
We should delete the whole `change.0` entry if the set is empty after removing `change.1`.